src/gluonts/model/wavenet/_network.py

```diff
@@ -378,70 +553,47 @@ def is_last_layer(i):
             end=None,
             axis=-1,
         )
-
-        tmp = F.slice_axis(
-            past_target, begin=-self.receptive_field, end=None, axis=-1
-        )
-        o = self.target_embed(tmp).swapaxes(1, 2)
-        o = F.concat(
-            o,
-            F.slice_axis(
+        queues = self.get_initial_conv_queues(
+            F,
+            past_target=F.slice_axis(
+                past_target, begin=-self.receptive_field, end=None, axis=-1
+            ),
+            features=F.slice_axis(
                 feature_slice, begin=-self.receptive_field, end=None, axis=-1
```

Comment:
Maybe there is a bug here in feature slicing. The features here are used to compute the queues for the first time step so they should be slice of `full_features` from `-self.pred_length - self.receptive_field + 1` to `-self.pred_length + 1`.

Currently they are sliced from `-self.receptive_field` to `None` of `feature_slice` which is same as the slice  
`full_features[-self.receptive_field:None]`. Is this intended?

Reply:
Yes, I think you are right.